### PR TITLE
feat(ui): Add some basic chart wrappers

### PR DESCRIPTION
**Pull request by billyvg**
* AreaChart w/ previous period
* PieChart

Data shape is expected to change, as well as a lot of defaults.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/areaChart.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import Tooltip from './components/tooltip';
+import XAxis from './components/xAxis';
+import YAxis from './components/yAxis';
+
+class AreaChart extends React.Component {
+  static propTypes = {
+    /**
+       * Height of chart
+       */
+    height: PropTypes.number,
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/charts/areaChart.jsx]: Is there a way to pass a width as well?

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+class BaseChart extends React.Component {
+  static propTypes = {
+    // see: https://ecomfe.github.io/echarts-doc/public/en/option.html
+    options: PropTypes.object,
+
+    // Chart height
+    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    // Chart width
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    renderer: PropTypes.oneOf(['canvas', 'svg']),
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/charts/baseChart.jsx]: Would we ever want canvas? Perhaps we shouldn't expose this prop at all

**Review by lynnagara — APPROVED** [review]
💯

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactEchartsCore from 'echarts-for-react/lib/core';
+import 'zrender/lib/svg/svg';
```
Comment (ayesha-omarali) [inline src/sentry/static/sentry/app/components/charts/baseChart.jsx]: what's this for? how's it used?

**Review by ayesha-omarali — APPROVED** [review]
🥇

File: src/sentry/static/sentry/app/components/charts/areaChart.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/areaChart.jsx]: Yeah I'll fix up the prop types.

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/baseChart.jsx]: I think when you have lots of datapoints you'll want to use `canvas` for performance reasons, I'll add a note in comments for this.

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/baseChart.jsx]: This is required for the svg renderer to work.

**Pull request merged**